content/moderation/logging.md

```diff
@@ -0,0 +1,118 @@
+
+### Moderation Actions
+
+Select moderation actions will generate message logs when used, and other actions allow you to enable logging if
+desired. A link for these logs is included in the mod log if mod log is enabled.
+
+#### Actions Generating Logs
+
+These actions immediately create a log of the last 100 messages in the channel.
+
+- Report
+- Timeout
+- Mute
+- Kick
+- Ban
+
+#### Actions Log Generation Optional
+
+These actions require logging to be enabled on their control panel.
+
+- Warn
+
+## Viewing Logs
+
+To view a message log after it has been created, you'll need to visit its link in a browser. YAGPDB cannot send message
+logs to a Discord channel, the only way to view them is online.
+
+- When a log is created, the link to view it follows the formula `/public/<server ID>/log/<log count>`. Note that this
+  makes it easy to scrape through all the logs created on a server. Assume any member that can view a log can view every
+  log on your server.
+- You can view a list of all message logs created on your server on the Logging page of the control panel.
+- When the `logs` command is executed, a link to the message log is sent as the response.
+- When a moderation action creates a log, a link is included in the mod log entry if enabled.
+
+When viewing a log, YAGPDB colors the usernames of each message author based on their roles at the time of log
+*viewing*, similarly to how they would appear on your server.
+
+## Access Control
+
+### Members can view message logs
+
+Selecting this mode on the control panel allows all members of your server to view any message logs created. Users will
+need to log in with their Discord account to view the message logs.
+
+#### Roles
+
+You may additionally choose to require roles for members to have in order to view message logs. Members with *one or
+more* of these roles will be permitted to view message logs.
+
+### Everyone can view message logs
+
+Selecting this mode on the control panel allows anyone with a link to a log to view all the logs on your server without
+needing to log in.
+
+## Deleted Messages
+
+A log contains any messages cached at the time of log creation. Deleted messages are cached for `1` hour before being
+removed. On [premium](/premium) servers, they are cached for `12` hours.
+
+In a message log, deleted messages are shown to users permitted to view them. If a user is not permitted to view deleted
+messages, they will not be able to see deleted messages, or that a message was deleted at all.
+
+## Blacklisting Channels
+
+Under the Logging page on the control panel you may blacklist channels for log creation. Using the logs command in a
+blacklisted channel will return an error message. Moderation actions will also not generate or attach logs in
+blacklisted channels.
+
+## Deleting Logs
+
+Users with write access to the control panel may delete individual logs or delete all logs on the server using the
+control panel.
+
+## Message Log Purge
```

Comment:
```suggestion
## Message Log Purge [relevant to self-hosters only]
```
I think it should be made more clear that this section is really only relevant to those self-hosting YAGPDB. The warning re: the official instance deleting logs after 30 days should be moved upward in the article.

Reply:
Personally I think it is better to move anything relevant to selfhosters
into its own category, which will mostly be just configuration values by
then. This way we can keep the main categories clean and focused on
end-user features. WDYT?

Nevertheless, the warning should definitely be further up on the page.

Reply:
I'm conflicted. I believe that the documentation should be for all users, including self hosters, hence why I've elected to include extra detail about inner workings of the bot in many places where the end-user wouldn't particularly care. On the other hand, I could see there being a section for end users and a section for hosters. I do think however that I personally prefer having the documentation of a feature thoroughly describe the feature's use and workings with detail and specificity. 

When looking at similar bots such as Red, their documentation is primarily aimed at the hoster with detailed descriptions of how the end user would use the features later on. Contrastingly, MEE6 documentation is composed of a series of tutorials walking the end user through setting up each plugin step by step. YAG docs fall somewhere in the middle, but the current version leans more towards the MEE6 docs than the Red docs, which I personally don't love. 

I think in an ideal world we have a full documentation and then quick start guides for end-users setting up plugins. Maybe expanding the learn page to all plugins of the bot rather than exclusively custom commands?

I think I could maybe see a subpage for self hosting on each relevant category, but I think putting self hosting anything in its own category, a), with subpages for each plugin would be messy and a waste of space, just mirroring the end user layout but under a self hosting category, or b), putting everything on a single page, requiring users to jump up and down a single page to get to information about several completely unrelated plugins just doesn' sit right with me. With either option we are forcing users to jump back and forth between the end user and self hoster's documentation for the same plugin.

+1 for moving the warning up.

Reply:
I will leave a more detailed reply later but do want to quickly mention there is a significant difference between Red and MEE6, YAGPDB, ... in that Red has no officially hosted instance and is primarily intended to be self-hosted.

Reply:
Yes, I think they are two extremes that Yag docs fall somewhere in between, but I think they each have relevant elements that apply to Yag and we should therefore examine them to see what does and doesn't work for us.

Reply:
Perhaps we can leave this info for selfhosters in for now, and discuss
the way going forward regarding integration of self-hosting docs in an
issue?

That way we don't unnecessarily block this PR; we can always change
things around at a later date.

Let me know what you think.

Reply:
Sounds good, but I think before any more pages are written we should come to a consensus about how the docs are going to be structured, since this is the second issue regarding the overall approach to documenting that's come up in the last week basically. It would be nice to come to consensus and maybe tighten up CONTRIBUTING.MD accordingly before we waste man-hours going back on pages we've already written.

Reply:
100% agreed. I will shortly (hah) open two separate issues with a
transcript of the relevant discussion to open up this PR for merging.

All other PRs are until resolution of these two issues blocked.